src/user.rs

```diff
@@ -13,10 +20,68 @@ use utils;
 /// 2. Add the user's name and public key to the protonfile
 ///
 /// Impure.
-pub fn new_user<P: AsRef<Path>>(public_key_path: P, name: String) -> Result<(), Error> {
-    let pub_key = try!(utils::file_as_string(public_key_path));
+pub fn new_user<P: AsRef<Path>>(public_key_path: P, name: &str) -> Result<(), Error> {
+    let pub_key = try!(get_public_key(public_key_path));
     let project = try!(utils::read_protonfile(None::<P>));
-    let new_project = try!(project.add_user(&name, &pub_key));
+    let new_project = try!(project.add_user(name, &pub_key));
     utils::write_protonfile(&new_project, None::<P>)
 }
 
+/// Identifies a user by their private SSH key by finding the
+/// corresponding public key in the project. This private key
+/// acts like the user's password, and should be protected.
+/// 
+/// Impure.
+pub fn id_user<P: AsRef<Path>>(private_key_path: P) -> Result<User, Error> {
+    let test_data: &[u8] = b"Testing to find private/public key pair";
+    
+    let mut private_key_file = try!(File::open(&private_key_path).map_err(Error::Io));
+    
+    let private_key = try!(openssl_RSA::private_key_from_pem(&mut private_key_file)
+        .map_err(ssl_error));
+
+    let signature = try!(private_key.sign(openssl_HashType::MD5, &test_data)
+        .map_err(ssl_error));
+
+    let project = try!(utils::read_protonfile(None::<P>));
+    for user in project.users {
+        let user_key = user.public_key.clone();
```

Comment:
Sort of the same deal as above. In order to return a User, and not &User, `user in project.users` has to be `user`, not `&user`, but it is then partially moved in lines 48/49 (String not Copyable), so it cannot be returned in line 56.